src/main/java/uk/ac/ic/wlgitbridge/bridge/lock/InProcessLockFactory.java

```diff
@@ -0,0 +1,37 @@
+package uk.ac.ic.wlgitbridge.bridge.lock;
+
+import uk.ac.ic.wlgitbridge.util.Log;
+
+import java.util.HashMap;
+
+public final class InProcessLockFactory implements ProjectLockFactory {
+  private final HashMap<String, InProcessLock> locks;
+  private final OperationTracker operationTracker;
+
+  public InProcessLockFactory() {
+    // TODO: Note for review: I'm not sure we need to keep these in a map,
```

Comment:
See my previous comment. I think the `HashMap` should be a `static` map within the `InProcessLock` class. That would make the behaviour of the two different lock classes the same.

Consider the following two cases:
```Java
InProcessLock lockA = new InProcessLock("wombat", operationTracker);
InProcessLock lockB = new InProcessLock("wombat", operationTracker);

lockA.lock();
lockB.lock();
```
In this case, both lockA and lockB would lock and continue, because they are not locking on any particular thing other than themselves. - They prevent thread-based collisions, but not another request coming along and modifying the same project at the same time. The project ID is only used for logging.

In the case of Postgres locks we have different behaviour:
```Java
PostgresLock lockA = new PostgresLock("wombat", operationTracker);
PostgresLock lockB = new PostgresLock("wombat", operationTracker);

lockA.lock();
lockB.lock();
```
Same logic, but we've now reached a deadlock, because `lockB` is also trying to acquire a lock on `lockA`'s project.. In this case, we *do* prevent two different HTTP requests operating on the same project at the same time.

I realise that we have protection against this here as we are getting locks out of the hashmap in the factory, but for consistency I think it is better to make the behaviour of the two lock classes the same.